File: lib/qt/AbilityListComboBox.cpp
/*
 * Copyright (c) 2016 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

#include <pkmn/qt/AbilityListComboBox.hpp>

#include <pkmn/database/lists.hpp>

namespace pkmn { namespace qt {

    AbilityListComboBox::AbilityListComboBox(
        int generation,
        QWidget* parent
    ): QComboBox(parent)
    {
        std::vector<std::string> abilities = pkmn::database::get_ability_list(
                                                 generation
                                             );
        for(auto iter = abilities.begin(); iter != abilities.end(); ++iter) {
            addItem(QString::fromStdString(*iter));
        }

        setEditable(false);
    }

}}

// Generated at build time
#include "../include/pkmn/qt/moc_AbilityListComboBox.cpp"

File: lib/swig/java/ComboBoxes/AbilityListComboBox.java
/*
 * Copyright (c) 2016 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

package nc.PKMN;

import javax.swing.JComboBox;

/**
 * A JComboBox filled with a list of abilities in a given generation.
 */
public class AbilityListComboBox extends JComboBox<String> {
    public AbilityListComboBox(int generation) {
        StringVector abilityList = Database.getAbilityList(generation);
        for(long i = 0; i < abilityList.size(); i++) {
            this.addItem(abilityList.get((int)i));
        }
        this.setEditable(false);
    }
}
